cudaextender/samples/sample_cudaextender.cpp

```diff
@@ -0,0 +1,184 @@
+        std::cout << "-d : Use Device Pointer API. If not provided uses Host Pointer API." << std::endl;
+        std::cout << "-h : Print help message." << std::endl;
+        std::exit(0);
+    }
+
+    // Fasta query and target files
+    const std::string target_file_path                   = std::string(CUDAEXTENDER_DATA_DIR) + "/sample.fa";
+    std::unique_ptr<io::FastaParser> fasta_parser_target = io::create_kseq_fasta_parser(target_file_path, 0, false);
+    // Assumes that only one sequence is present per file
+    const std::string target_sequence = fasta_parser_target->get_sequence_by_id(0).seq;
+
+    const std::string query_file_path                   = std::string(CUDAEXTENDER_DATA_DIR) + "/sample.fa";
+    std::unique_ptr<io::FastaParser> fasta_parser_query = io::create_kseq_fasta_parser(query_file_path, 0, false);
+    // Assumes that only one sequence is present per file
+    const std::string query_sequence = fasta_parser_query->get_sequence_by_id(0).seq;
+
+    // CSV SeedPairs file - Each row -> query_position_in_read_,
+    // target_position_in_read_
+    const std::string seed_pairs_file_path = std::string(CUDAEXTENDER_DATA_DIR) + "/sample_seed_pairs.csv";
+
+    //TODO - pinned seed_pairs
+    std::vector<SeedPair> h_seed_pairs;
+    // Following function loops through all seed_pairs in the sample_seed_pairs.csv and returns
+    // results in the passed vector
+    parse_seed_pairs(seed_pairs_file_path, h_seed_pairs);
+    std::cerr << "Number of Seed Pairs: " << h_seed_pairs.size() << std::endl;
+
+    // Define Scoring Matrix
+    const int32_t score_matrix[NUC2] = {91, -114, -31, -123, -1000, -1000, -100, -9100,
+                                        -114, 100, -125, -31, -1000, -1000, -100, -9100,
+                                        -31, -125, 100, -114, -1000, -1000, -100, -9100,
+                                        -123, -31, -114, 91, -1000, -1000, -100, -9100,
+                                        -1000, -1000, -1000, -1000, -1000, -1000, -1000, -9100,
+                                        -1000, -1000, -1000, -1000, -1000, -1000, -1000, -9100,
+                                        -100, -100, -100, -100, -1000, -1000, -100, -9100,
+                                        -9100, -9100, -9100, -9100, -9100, -9100, -9100, -9100};
+
+    // Allocate pinned memory for query and target strings
+    pinned_host_vector<int8_t> h_encoded_target(target_sequence.length());
+    pinned_host_vector<int8_t> h_encoded_query(target_sequence.length());
+
+    encode_sequence(h_encoded_target.data(), target_sequence.c_str(), target_sequence.length());
+    encode_sequence(h_encoded_query.data(), query_sequence.c_str(), query_sequence.length());
+    // Create a stream for async use
+    CudaStream stream0 = make_cuda_stream();
+    // Create an allocator for use with both APIs
+    const std::size_t max_gpu_memory = cudautils::find_largest_contiguous_device_memory_section();
```

Comment:
`find_largest_contiguous_device_memory_section` returns an `int64_t` instead of `size_t`.

Reply:
Done.